python/cudf/cudf/core/series.py

```diff
@@ -5200,14 +5196,29 @@ def value_counts(
                 index=cudf.Index([], dtype=self.dtype),
             )
 
-        res = self.groupby(self, dropna=dropna).count(dropna=dropna)
+        if bins is not None:
+            res = self.groupby(series_bins, dropna=dropna).count(dropna=dropna)
+            res = res[res.index.notna()]
+            res = res.sort_index()
```

Comment:
Why do we need this change? I don't think this should be necessary here.

Reply:
This is what pandas does in their code and it changes the order of the index. If we don't do that then my tests fail because we aren't matching the panda's index.

Reply:
If you think we should leave it out, we'd need to figure out how to test in a way that ignores the order of the index, which might be challenging to do. What do you think?

Reply:
@galipremsagar it looks like this was the main open question, any thoughts on how we should proceed?

Reply:
commented below..